src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -597,6 +597,13 @@ class Typer extends Namer with TypeAssigner with Applications with Implicits wit
         untpd.TypeTree(defn.FunctionClass(args.length).typeRef), args :+ body), pt)
     else {
       val params = args.asInstanceOf[List[untpd.ValDef]]
+
+      pt match {
+        case pt: TypeVar if untpd.isFunctionWithImplicitParamType(tree) =>
+          isFullyDefined(pt, ForceDegree.noBottom)
```

Comment:
A comment should clarify that this method is called for its side-effects, and why we don't need to do anything special when it return `false` (I think this is the only instance of `isFullyDefined` in the compiler where we ignore the return value)